core/module_shared.h

```diff
@@ -413,7 +413,9 @@ typedef struct _privmod_t {
     char path[MAXIMUM_PATH];
     uint ref_count;
     bool externally_loaded;
-    bool is_client; /* or Extension */
+    /* XXX: is_top_level_client could perhaps replace is_client */
```

Comment:
nit: Use capitalized and end-punctuated sentences in new comments.  It would also be good to elaborate and say something like "if is_client is only ever used for top-level clients: the current code may need some auditing and cleanup to decide, as there is an attempt to set is_client  in privload_process_imports() but it is suspiciously inside a lookup failure conditional and so is not reached."

It would also be good to add a corresponding XXX comment at that point where is_client is set to true but only inside the if(), which looks suspect.

This will all help future cleanup not have to puzzle over why this is "could perhaps".

Maybe a new issue could be filed and cited in these XXX as well.